Index: file-method.c
===================================================================
RCS file: /cvs/gnome/gnome-vfs/modules/file-method.c,v
retrieving revision 1.61
diff -u -u -r1.61 file-method.c
--- file-method.c	2000/09/05 16:27:54	1.61
+++ file-method.c	2000/09/16 21:47:24
@@ -159,6 +159,8 @@
 	_GNOME_VFS_METHOD_PARAM_CHECK (method_handle != NULL);
 	_GNOME_VFS_METHOD_PARAM_CHECK (uri != NULL);
 
+	sleep (5);
+
 	if (mode & GNOME_VFS_OPEN_READ) {
 		if (mode & GNOME_VFS_OPEN_WRITE)
 			unix_mode = O_RDWR;
@@ -297,6 +299,8 @@
 
 	g_return_val_if_fail (method_handle != NULL, GNOME_VFS_ERROR_INTERNAL);
 
+	sleep (5);
+
 	file_handle = (FileHandle *) method_handle;
 
 	do
@@ -558,6 +562,8 @@
 	size = 256;
 	buffer = g_malloc (size);
           
+	sleep (5);
+
 	while (1) {
 		int read_size;
 
@@ -583,6 +589,8 @@
 	struct stat statbuf;
 	gboolean followed_symlink;
 
+	sleep (5);
+
 	if (statptr == NULL) {
 		statptr = &statbuf;
 	}
@@ -634,6 +642,8 @@
 {
 	struct stat statbuf;
 
+	sleep (5);
+
 	if (statptr == NULL)
 		statptr = &statbuf;
 
@@ -658,6 +668,8 @@
 	gchar *directory_name;
 	DIR *dir;
 
+	sleep (5);
+
 	directory_name = get_path_from_uri (uri);
 	if (directory_name == NULL)
 		return GNOME_VFS_ERROR_INVALID_URI;
@@ -708,6 +720,8 @@
            once.  */
 	filter_called = FALSE;
 
+	sleep (5);
+
 	filter = handle->filter;
 	if (filter != NULL) {
 		filter_needs = gnome_vfs_directory_filter_get_needs (filter);
@@ -791,6 +805,8 @@
 {
 	GnomeVFSResult result;
 	gboolean skip;
+
+	sleep (5);
 
 	do {
 		result = read_directory ((DirectoryHandle *) method_handle,
