Index: .cvsignore
===================================================================
RCS file: /cvs/gnome/gnome-vfs/.cvsignore,v
retrieving revision 1.4
diff -u -u -p -r1.4 .cvsignore
--- .cvsignore	2000/08/31 11:32:38	1.4
+++ .cvsignore	2000/10/15 21:54:14
@@ -28,3 +28,4 @@ vfsConf.sh
 vfscorbaConf.sh
 vfspthreadConf.sh
 gnome-vfs-config
+gnomevfs-2.0.pc
Index: Makefile.am
===================================================================
RCS file: /cvs/gnome/gnome-vfs/Makefile.am,v
retrieving revision 1.16
diff -u -u -p -r1.16 Makefile.am
--- Makefile.am	2000/10/11 17:36:39	1.16
+++ Makefile.am	2000/10/15 21:54:14
@@ -31,7 +31,8 @@ EXTRA_DIST =					\
 	vfsConf.sh.in				\
 	vfscorbaConf.sh.in			\
 	vfspthreadConf.sh.in			\
-	gnome-vfs-config.in
+	gnome-vfs-config.in			\
+	gnomevfs-2.0.pc
 
 bin_SCRIPTS = $(top_builddir)/gnome-vfs-config
 
@@ -82,4 +83,6 @@ gnome-vfs-config: gnome-vfs-config.in Ma
 	  && mv gnome-vfs-config.tmp gnome-vfs-config && \
 		chmod 755 gnome-vfs-config
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = gnomevfs-2.0.pc
 
Index: configure.in
===================================================================
RCS file: /cvs/gnome/gnome-vfs/configure.in,v
retrieving revision 1.79
diff -u -u -p -r1.79 configure.in
--- configure.in	2000/10/11 17:36:39	1.79
+++ configure.in	2000/10/15 21:54:15
@@ -7,6 +7,7 @@ AM_MAINTAINER_MODE
 
 AC_ISC_POSIX
 AC_PROG_CC
+AC_PROG_CPP
 AC_PROG_AWK
 AM_PROG_LIBTOOL
 
@@ -34,7 +35,7 @@ dnl = Profiling support
 dnl ====================================
 
 dnl Version of the GNOME Platform (1002000 for GNOME 1.2.x)
-GNOME_PLATFORM_VERSION=1002000
+GNOME_PLATFORM_VERSION=1095000
 AC_DEFINE_UNQUOTED(GNOME_PLATFORM_VERSION, $GNOME_PLATFORM_VERSION)
 AM_CONDITIONAL(PLATFORM_GNOME_2, test $GNOME_PLATFORM_VERSION -ge 1095000)
 
@@ -91,13 +92,13 @@ AC_ARG_WITH(default-backend,
 	      [  --with-default-backend=(corba|pthread)		    ],
 	      default_backend="$withval")
 
-AM_PATH_GLIB(1.2.0,,
+AM_PATH_GLIB_2_0(1.3.0,,
   AC_MSG_ERROR([
-*** GLIB 1.2.0 or better is required. The latest version of GLIB
+*** GLIB 1.3.0 or better is required. The latest version of GLIB
 *** is always available from ftp://ftp.gtk.org/.]),
     gmodule gthread)
 
-AM_PATH_GTK(1.2.0,,
+AM_PATH_GTK_2_0(1.3.0,,
   AC_MSG_ERROR([
 *** Gtk+ 1.2.0 or better is required. The latest version of Gtk+
 *** is always available from ftp://ftp.gtk.org/.]),
@@ -436,6 +437,7 @@ po/Makefile.in
 test/Makefile
 test/gnome-file-selection/Makefile
 test/vfs-run
+gnomevfs-2.0.pc
 ])
 
 chmod u+x test/vfs-run
Index: gnomevfs-2.0.pc.in
===================================================================
RCS file: gnomevfs-2.0.pc.in
diff -N gnomevfs-2.0.pc.in
--- /dev/null	Tue May  5 16:32:27 1998
+++ gnomevfs-2.0.pc.in	Sun Oct 15 17:54:15 2000
@@ -0,0 +1,12 @@
+prefix=/gnome/head/INSTALL
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include
+
+
+Name: gnome-vfs
+Description: The GNOME virtual file-system libraries
+Version: @VERSION@
+Requires: oaf-2.0,gthread-2.0,gmodule-2.0
+Libs: -L${libdir} -lgnomevfs
+Cflags: -I${includedir} -I${libdir}/vfs/include
