? oaf-0.6.1.tar.gz
? oaf-2.6.1.tar.gz
Index: .cvsignore
===================================================================
RCS file: /cvs/gnome/oaf/.cvsignore,v
retrieving revision 1.7
diff -u -u -r1.7 .cvsignore
--- .cvsignore	2000/06/19 16:08:10	1.7
+++ .cvsignore	2000/12/13 10:47:36
@@ -23,3 +23,4 @@
 po
 stamp-h
 stamp-h.in
+oaf-2.0.pc
Index: ChangeLog
===================================================================
RCS file: /cvs/gnome/oaf/ChangeLog,v
retrieving revision 1.127
diff -u -u -r1.127 ChangeLog
--- ChangeLog	2000/11/25 23:18:09	1.127
+++ ChangeLog	2000/12/13 10:47:38
@@ -1,3 +1,12 @@
+2000-11-22  Martin Baulig  <martin@home-of-linux.org>
+
+	* autogen.sh: Use gnome-common with the GNOME 2 macros.
+
+	* configure.in: Use GNOME_PKGCONFIG_CHECK_MODULES to check for
+	GLib 1.3.x, ORBit and libxml2. Use AM_GNOME2_GETTEXT instead of
+	AM_GNU_OAF_GETTEXT. Call GNOME_PLATFORM_GNOME_2(yes, force). Use
+	GNOME_GTKDOC_CHECK to check for gtk-doc.
+
 2000-11-24  Michael Meeks  <michael@helixcode.com>
 
 	* oafd/od-utils.c (od_utils_load_config_file): fix leak.
Index: Makefile.am
===================================================================
RCS file: /cvs/gnome/oaf/Makefile.am,v
retrieving revision 1.12
diff -u -u -r1.12 Makefile.am
--- Makefile.am	2000/08/10 21:55:58	1.12
+++ Makefile.am	2000/12/13 10:47:38
@@ -28,6 +28,7 @@
 	ChangeLog 		\
 	NEWS 			\
 	README			\
+	oaf-2.0.pc.in		\
 	$(NULL)
 
 oafconfigdir=$(sysconfdir)/oaf
@@ -41,6 +42,9 @@
 
 m4datadir=$(datadir)/aclocal
 m4data_DATA=oaf.m4
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = oaf-2.0.pc
 
 #dist-hook: oaf.spec
 #	cp oaf.spec $(distdir)
Index: autogen.sh
===================================================================
RCS file: /cvs/gnome/oaf/autogen.sh,v
retrieving revision 1.3
diff -u -u -r1.3 autogen.sh
--- autogen.sh	2000/08/31 09:23:39	1.3
+++ autogen.sh	2000/12/13 10:47:38
@@ -4,77 +4,12 @@
 srcdir=`dirname $0`
 test -z "$srcdir" && srcdir=.
 
-ORIGDIR=`pwd`
-cd $srcdir
+PKG_NAME="oaf"
 
-PROJECT=oaf
-TEST_TYPE=-f
-FILE=idl/oaf.idl
-
-DIE=0
-
-(autoconf --version) < /dev/null > /dev/null 2>&1 || {
-	echo
-	echo "You must have autoconf installed to compile $PROJECT."
-	echo "Download the appropriate package for your distribution,"
-	echo "or get the source tarball at ftp://ftp.gnu.org/pub/gnu/"
-	DIE=1
-}
-
-(libtool --version) < /dev/null > /dev/null 2>&1 || {
-	echo
-	echo "You must have libtool installed to compile $PROJECT."
-	echo "Get ftp://alpha.gnu.org/gnu/libtool-1.0h.tar.gz"
-	echo "(or a newer version if it is available)"
-	DIE=1
-}
-
-(automake --version) < /dev/null > /dev/null 2>&1 || {
-	echo
-	echo "You must have automake installed to compile $PROJECT."
-	echo "Get ftp://ftp.cygnus.com/pub/home/tromey/automake-1.2d.tar.gz"
-	echo "(or a newer version if it is available)"
-	DIE=1
-}
-
-if test "$DIE" -eq 1; then
-	exit 1
-fi
-
-test $TEST_TYPE $FILE || {
-	echo "You must run this script in the top-level $PROJECT directory"
+(test -f idl/oaf.idl) || {
+	echo "You must run this script in the top-level oaf directory"
 	exit 1
 }
-
-if test -z "$*"; then
-	echo "I am going to run ./configure with no arguments - if you wish "
-        echo "to pass any to it, please specify them on the $0 command line."
-fi
-
-case $CC in
-*lcc | *lcc\ *) am_opt=--include-deps;;
-esac
-
-echo "Running gettextize...  Ignore non-fatal messages."
-# Hmm, we specify --force here, since otherwise things don't
-# get added reliably, but we don't want to overwrite intl
-# while making dist.
-echo "no" | gettextize --copy --force
-
-echo "Running libtoolize"
-libtoolize --copy --force
-
-aclocal $ACLOCAL_FLAGS
-
-# optionally feature autoheader
-(autoheader --version)  < /dev/null > /dev/null 2>&1 && autoheader
-
-automake -a $am_opt
-autoconf
-
-cd $ORIGDIR
 
-$srcdir/configure --enable-maintainer-mode "$@"
+USE_GNOME2_MACROS=1 . gnome-autogen.sh
 
-echo 
-echo "Now type 'make' to compile $PROJECT."
Index: configure.in
===================================================================
RCS file: /cvs/gnome/oaf/configure.in,v
retrieving revision 1.49
diff -u -u -r1.49 configure.in
--- configure.in	2000/11/12 15:49:08	1.49
+++ configure.in	2000/12/13 10:47:38
@@ -1,15 +1,32 @@
 dnl Process this file with autoconf to produce a configure script.
 AC_INIT(idl/oaf.idl)
 
-OAF_MAJOR_VERSION=0
+OAF_MAJOR_VERSION=2
 OAF_MINOR_VERSION=6
 OAF_MICRO_VERSION=1
+OAF_INTERFACE_AGE=0
+OAF_BINARY_AGE=0
 OAF_VERSION="$OAF_MAJOR_VERSION.$OAF_MINOR_VERSION.$OAF_MICRO_VERSION"
 AM_INIT_AUTOMAKE(oaf, $OAF_VERSION)
 AC_SUBST(OAF_MAJOR_VERSION)
 AC_SUBST(OAF_MINOR_VERSION)
 AC_SUBST(OAF_MICRO_VERSION)
+AC_SUBST(OAF_INTERFACE_AGE)
+AC_SUBST(OAF_BINARY_AGE)
 
+# libtool versioning
+LT_RELEASE=$OAF_MAJOR_VERSION.$OAF_MINOR_VERSION
+LT_CURRENT=`expr $OAF_MICRO_VERSION - $OAF_INTERFACE_AGE`
+LT_REVISION=$OAF_INTERFACE_AGE
+LT_AGE=`expr $OAF_BINARY_AGE - $OAF_INTERFACE_AGE`
+LT_VERSION_INFO='-version-info ${LT_CURRENT}:${LT_REVISION}:${LT_AGE} -release ${LT_RELEASE}'
+AC_SUBST(LT_VERSION_INFO)
+AC_SUBST(LT_RELEASE)
+AC_SUBST(LT_CURRENT)
+AC_SUBST(LT_REVISION)
+AC_SUBST(LT_AGE)
+
+
 dnl Specify a header configuration file
 AM_CONFIG_HEADER(config.h)
 
@@ -34,75 +51,30 @@
 
 dnl internationalization support
 ALL_LINGUAS="da de es fi fr ga gl hu it ja ko no ru sl sv tr uk pl"
-AM_GNU_OAF_GETTEXT
+AM_GNOME2_GETTEXT
 
+GNOME_COMMON_INIT
+GNOME_PLATFORM_GNOME_2(yes, force)
 
 dnl Checks for libraries.
-AM_PATH_GLIB(1.2.0,,
-AC_MSG_ERROR([
-*** GLIB 1.2.0 or better is required. The latest version of GLIB
-*** is always available from ftp://ftp.gtk.org/.]), gmodule)
-
-AM_PATH_ORBIT(0.5.1,,AC_MSG_ERROR([ORBit not installed or installation problem]))
+GNOME_PKGCONFIG_CHECK_MODULES(GLIB, gmodule-2.0:1.3.1)
+AC_SUBST(GLIB_LIBS)
+AC_SUBST(GLIB_CFLAGS)
+
+GNOME_PKGCONFIG_CHECK_MODULES(ORBIT, ORBit-2.0:0.5.3)
+AC_SUBST(ORBIT_LIBS)
+AC_SUBST(ORBIT_CFLAGS)
+AC_PATH_PROG(ORBIT_IDL, orbit-idl)
 AC_DEFINE(HAVE_ORB_ORBIT_H)
 
-dnl Check for ORBit > 0.5.1 (CORBA_Object_create_request is prototyped
-dnl in orb/dii.h in the "new" ORBit, but not ORBit 0.5.1, so #defining
-dnl it to nothing results in a compile error iff we have the new ORBit.)
-saved_CFLAGS="$CFLAGS"
-saved_LDFLAGS="$LDFLAGS"
-CFLAGS="$CFLAGS $ORBIT_CFLAGS"
-LDFLAGS="$LDFLAGS $ORBIT_LDFLAGS"
-AC_TRY_COMPILE([
-#define CORBA_Object_create_request
-#include <orb/dii.h>
-],[], AC_MSG_ERROR([You need a newer version of ORBit. See the README.]), :)
-CFLAGS="$saved_CFLAGS"
-LDFLAGS="$saved_LDFLAGS"
-
-APP_LIBS=""
-AC_SUBST(APP_LIBS)
-
-dnl	
-dnl Locate the xml library
-dnl
-AC_CHECK_PROG(XML_CONFIG, xml-config, xml-config)
-if test "$XML_CONFIG" != "" 
-then
-  XML_LIBS="`$XML_CONFIG --libs`"
-  XML_CFLAGS="`$XML_CONFIG --cflags`"
-else
-  AC_MSG_ERROR(Can't find gnome-xml library.)
-fi
+GNOME_PKGCONFIG_CHECK_MODULES(XML, libxml-2.0:2.2.8)
 AC_SUBST(XML_LIBS)
 AC_SUBST(XML_CFLAGS)
 
 dnl ======================
 dnl = gtk-doc stuff
 dnl ======================
-
-AC_CHECK_PROG(GTKDOC, gtkdoc-mkdb, true, false)
-AM_CONDITIONAL(HAVE_GTK_DOC, $GTKDOC)
-AC_SUBST(HAVE_GTK_DOC)
-
-dnl Let people disable the gtk-doc stuff.
-AC_ARG_ENABLE(gtk-doc, [  --enable-gtk-doc  Use gtk-doc to build documentation [default=auto]], enable_gtk_doc="$enableval", enable_gtk_doc=auto)
-
-if test x$enable_gtk_doc = xauto ; then
-  if test x$GTKDOC = xtrue ; then
-    enable_gtk_doc=yes
-  else
-    enable_gtk_doc=no
-  fi
-fi
-dnl NOTE: We need to use a separate automake conditional for this
-dnl       to make this work with the tarballs.
-AM_CONDITIONAL(ENABLE_GTK_DOC, test x$enable_gtk_doc = xyes)
-
-dnl ======================
-dnl = end gtk-doc stuff
-dnl ======================
-
+GNOME_GTKDOC_CHECK
 
 AC_CHECK_LIB(popt, poptGetContext,,
 AC_MSG_ERROR([You must have popt installed. Get it from ftp://ftp.redhat.com/pub/redhat/code/popt/]))
@@ -184,5 +156,6 @@
 po/Makefile.in
 oaf-config
 oafConf.sh
+oaf-2.0.pc
 ],
 [sed -e "/POTFILES =/r po/POTFILES" po/Makefile.in > po/Makefile])
Index: oaf-2.0.pc.in
===================================================================
RCS file: oaf-2.0.pc.in
diff -N oaf-2.0.pc.in
--- /dev/null	Tue May  5 16:32:27 1998
+++ oaf-2.0.pc.in	Wed Dec 13 05:47:38 2000
@@ -0,0 +1,12 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+
+Name: OAF
+Version: @VERSION@
+Description: Object activation framework for GNOME
+Requires: ORBit-2.0
+Libs: -L${libdir} -loaf
+Cflags: -I${includedir}
Index: liboaf/Makefile.am
===================================================================
RCS file: /cvs/gnome/oaf/liboaf/Makefile.am,v
retrieving revision 1.19
diff -u -u -r1.19 Makefile.am
--- liboaf/Makefile.am	2000/10/06 04:25:33	1.19
+++ liboaf/Makefile.am	2000/12/13 10:47:38
@@ -15,6 +15,7 @@
 LDFLAGS=		\
 	@ORBIT_LIBS@ 	\
 	@GLIB_LIBS@	\
+	@LT_VERSION_INFO@	\
 	$(NULL)
 
 liboafincludedir = $(includedir)/liboaf
Index: liboaf/oaf-mainloop.c
===================================================================
RCS file: /cvs/gnome/oaf/liboaf/oaf-mainloop.c,v
retrieving revision 1.27
diff -u -u -r1.27 oaf-mainloop.c
--- liboaf/oaf-mainloop.c	2000/11/16 23:08:08	1.27
+++ liboaf/oaf-mainloop.c	2000/12/13 10:47:39
@@ -50,9 +50,9 @@
 
 #include <orb/orbit.h>
 
-static int oaf_corba_prio = G_PRIORITY_LOW;
-
 #ifndef ORBIT_USES_GLIB_MAIN_LOOP
+
+static int oaf_corba_prio = G_PRIORITY_LOW;
 
 static gboolean
 orb_handle_connection (GIOChannel * source, GIOCondition cond,
Index: oafd/main.c
===================================================================
RCS file: /cvs/gnome/oaf/oafd/main.c,v
retrieving revision 1.22
diff -u -u -r1.22 main.c
--- oafd/main.c	2000/11/09 17:49:59	1.22
+++ oafd/main.c	2000/12/13 10:47:39
@@ -109,7 +109,7 @@
 
 	poptFreeContext (ctx);
 
-        LIBXML_TEST_VERSION
+        LIBXML_TEST_VERSION;
 
 	ml = g_main_new (FALSE);
 
Index: oafd/od-load.c
===================================================================
RCS file: /cvs/gnome/oaf/oafd/od-load.c,v
retrieving revision 1.19
diff -u -u -r1.19 od-load.c
--- oafd/od-load.c	2000/11/09 17:49:59	1.19
+++ oafd/od-load.c	2000/12/13 10:47:39
@@ -53,7 +53,7 @@
 static void
 od_entry_read_props (OAF_ServerInfo * ent, xmlNodePtr node)
 {
-	int i, n;
+	int n;
 	xmlNodePtr sub;
 	OAF_Property *curprop;
 
@@ -71,9 +71,12 @@
 	ent->props._length = n;
 	ent->props._buffer = curprop = g_new (OAF_Property, n);
 
-	for (i = 0, sub = node->xmlChildrenNode; i < n; sub = sub->next, i++) {
+	for (sub = node->xmlChildrenNode; sub; sub = sub->next) {
 		char *type, *valuestr;
 
+		if (sub->type != XML_ELEMENT_NODE)
+			continue;
+
 		type = xmlGetProp (sub, "type");
 		if (!type)
 			continue;
@@ -110,8 +113,10 @@
 			curprop->v._u.value_stringv._buffer =
 				CORBA_sequence_CORBA_string_allocbuf (o);
 
-			for (j = 0, sub2 = sub->xmlChildrenNode; j < o;
-			     sub2 = sub2->next, j++) {
+			for (j = 0, sub2 = sub->xmlChildrenNode; sub2;
+			     sub2 = sub2->next) {
+                                if (sub2->type != XML_ELEMENT_NODE)
+                                        continue;
 				valuestr = xmlGetProp (sub2, "value");
 				if (valuestr)
 					curprop->v._u.
@@ -126,6 +131,7 @@
 						CORBA_string_dup ("");
 				}
 				free (valuestr);
+                                j++;
 			}
 
 		} else if (!strcasecmp (type, "number")) {
Index: po/Makefile.in.in
===================================================================
RCS file: Makefile.in.in
diff -N Makefile.in.in
--- /dev/null	Tue May  5 16:32:27 1998
+++ Makefile.in.in	Wed Dec 13 05:47:40 2000
@@ -0,0 +1,248 @@
+# Makefile for program source directory in GNU NLS utilities package.
+# Copyright (C) 1995, 1996, 1997 by Ulrich Drepper <drepper@gnu.ai.mit.edu>
+#
+# This file file be copied and used freely without restrictions.  It can
+# be used in projects which are not available under the GNU Public License
+# but which still want to provide support for the GNU gettext functionality.
+# Please note that the actual code is *not* freely available.
+
+PACKAGE = @PACKAGE@
+VERSION = @VERSION@
+
+SHELL = /bin/sh
+@SET_MAKE@
+
+srcdir = @srcdir@
+top_srcdir = @top_srcdir@
+VPATH = @srcdir@
+
+prefix = @prefix@
+exec_prefix = @exec_prefix@
+datadir = $(prefix)/@DATADIRNAME@
+localedir = $(datadir)/locale
+gnulocaledir = $(prefix)/share/locale
+gettextsrcdir = $(prefix)/share/gettext/po
+subdir = po
+
+INSTALL = @INSTALL@
+INSTALL_DATA = @INSTALL_DATA@
+MKINSTALLDIRS = $(top_srcdir)/@MKINSTALLDIRS@
+
+CC = @CC@
+GENCAT = @GENCAT@
+GMSGFMT = PATH=../src:$$PATH @GMSGFMT@
+MSGFMT = @MSGFMT@
+XGETTEXT = PATH=../src:$$PATH @XGETTEXT@
+MSGMERGE = PATH=../src:$$PATH msgmerge
+
+DEFS = @DEFS@
+CFLAGS = @CFLAGS@
+CPPFLAGS = @CPPFLAGS@
+
+INCLUDES = -I.. -I$(top_srcdir)/intl
+
+COMPILE = $(CC) -c $(DEFS) $(INCLUDES) $(CPPFLAGS) $(CFLAGS) $(XCFLAGS)
+
+SOURCES = cat-id-tbl.c
+POFILES = @POFILES@
+GMOFILES = @GMOFILES@
+DISTFILES = ChangeLog Makefile.in.in POTFILES.in $(PACKAGE).pot \
+stamp-cat-id $(POFILES) $(GMOFILES) $(SOURCES)
+
+POTFILES = \
+
+CATALOGS = @CATALOGS@
+CATOBJEXT = @CATOBJEXT@
+INSTOBJEXT = @INSTOBJEXT@
+
+.SUFFIXES:
+.SUFFIXES: .c .o .po .pox .gmo .mo .msg .cat
+
+.c.o:
+	$(COMPILE) $<
+
+.po.pox:
+	$(MAKE) $(PACKAGE).pot
+	$(MSGMERGE) $< $(srcdir)/$(PACKAGE).pot -o $*.pox
+
+.po.mo:
+	$(MSGFMT) -o $@ $<
+
+.po.gmo:
+	file=$(srcdir)/`echo $* | sed 's,.*/,,'`.gmo \
+	  && rm -f $$file && $(GMSGFMT) -o $$file $<
+
+.po.cat:
+	sed -f ../intl/po2msg.sed < $< > $*.msg \
+	  && rm -f $@ && $(GENCAT) $@ $*.msg
+
+
+all: all-@USE_NLS@
+
+all-yes: cat-id-tbl.c $(CATALOGS)
+all-no:
+
+$(srcdir)/$(PACKAGE).pot: $(POTFILES)
+	$(XGETTEXT) --default-domain=$(PACKAGE) --directory=$(top_srcdir) \
+	  --add-comments --keyword=_ --keyword=N_ \
+	  --files-from=$(srcdir)/POTFILES.in \
+	&& test ! -f $(PACKAGE).po \
+	   || ( rm -f $(srcdir)/$(PACKAGE).pot \
+		&& mv $(PACKAGE).po $(srcdir)/$(PACKAGE).pot )
+
+$(srcdir)/cat-id-tbl.c: stamp-cat-id; @:
+$(srcdir)/stamp-cat-id: $(PACKAGE).pot
+	rm -f cat-id-tbl.tmp
+	sed -f ../intl/po2tbl.sed $(srcdir)/$(PACKAGE).pot \
+		| sed -e "s/@PACKAGE NAME@/$(PACKAGE)/" > cat-id-tbl.tmp
+	if cmp -s cat-id-tbl.tmp $(srcdir)/cat-id-tbl.c; then \
+	  rm cat-id-tbl.tmp; \
+	else \
+	  echo cat-id-tbl.c changed; \
+	  rm -f $(srcdir)/cat-id-tbl.c; \
+	  mv cat-id-tbl.tmp $(srcdir)/cat-id-tbl.c; \
+	fi
+	cd $(srcdir) && rm -f stamp-cat-id && echo timestamp > stamp-cat-id
+
+
+install: install-exec install-data
+install-exec:
+install-data: install-data-@USE_NLS@
+install-data-no: all
+install-data-yes: all
+	if test -r "$(MKINSTALLDIRS)"; then \
+	  $(MKINSTALLDIRS) $(datadir); \
+	else \
+	  $(SHELL) $(top_srcdir)/mkinstalldirs $(datadir); \
+	fi
+	@catalogs='$(CATALOGS)'; \
+	for cat in $$catalogs; do \
+	  cat=`basename $$cat`; \
+	  case "$$cat" in \
+	    *.gmo) destdir=$(gnulocaledir);; \
+	    *)     destdir=$(localedir);; \
+	  esac; \
+	  lang=`echo $$cat | sed 's/\$(CATOBJEXT)$$//'`; \
+	  dir=$$destdir/$$lang/LC_MESSAGES; \
+	  if test -r "$(MKINSTALLDIRS)"; then \
+	    $(MKINSTALLDIRS) $$dir; \
+	  else \
+	    $(SHELL) $(top_srcdir)/mkinstalldirs $$dir; \
+	  fi; \
+	  if test -r $$cat; then \
+	    $(INSTALL_DATA) $$cat $$dir/$(PACKAGE)$(INSTOBJEXT); \
+	    echo "installing $$cat as $$dir/$(PACKAGE)$(INSTOBJEXT)"; \
+	  else \
+	    $(INSTALL_DATA) $(srcdir)/$$cat $$dir/$(PACKAGE)$(INSTOBJEXT); \
+	    echo "installing $(srcdir)/$$cat as" \
+		 "$$dir/$(PACKAGE)$(INSTOBJEXT)"; \
+	  fi; \
+	  if test -r $$cat.m; then \
+	    $(INSTALL_DATA) $$cat.m $$dir/$(PACKAGE)$(INSTOBJEXT).m; \
+	    echo "installing $$cat.m as $$dir/$(PACKAGE)$(INSTOBJEXT).m"; \
+	  else \
+	    if test -r $(srcdir)/$$cat.m ; then \
+	      $(INSTALL_DATA) $(srcdir)/$$cat.m \
+		$$dir/$(PACKAGE)$(INSTOBJEXT).m; \
+	      echo "installing $(srcdir)/$$cat as" \
+		   "$$dir/$(PACKAGE)$(INSTOBJEXT).m"; \
+	    else \
+	      true; \
+	    fi; \
+	  fi; \
+	done
+	if test "$(PACKAGE)" = "gettext"; then \
+	  if test -r "$(MKINSTALLDIRS)"; then \
+	    $(MKINSTALLDIRS) $(gettextsrcdir); \
+	  else \
+	    $(SHELL) $(top_srcdir)/mkinstalldirs $(gettextsrcdir); \
+	  fi; \
+	  $(INSTALL_DATA) $(srcdir)/Makefile.in.in \
+			  $(gettextsrcdir)/Makefile.in.in; \
+	else \
+	  : ; \
+	fi
+
+# Define this as empty until I found a useful application.
+installcheck:
+
+uninstall:
+	catalogs='$(CATALOGS)'; \
+	for cat in $$catalogs; do \
+	  cat=`basename $$cat`; \
+	  lang=`echo $$cat | sed 's/\$(CATOBJEXT)$$//'`; \
+	  rm -f $(localedir)/$$lang/LC_MESSAGES/$(PACKAGE)$(INSTOBJEXT); \
+	  rm -f $(localedir)/$$lang/LC_MESSAGES/$(PACKAGE)$(INSTOBJEXT).m; \
+	  rm -f $(gnulocaledir)/$$lang/LC_MESSAGES/$(PACKAGE)$(INSTOBJEXT); \
+	  rm -f $(gnulocaledir)/$$lang/LC_MESSAGES/$(PACKAGE)$(INSTOBJEXT).m; \
+	done
+	rm -f $(gettextsrcdir)/po-Makefile.in.in
+
+check: all
+
+cat-id-tbl.o: ../intl/libgettext.h
+
+dvi info tags TAGS ID:
+
+mostlyclean:
+	rm -f core core.* *.pox $(PACKAGE).po *.old.po cat-id-tbl.tmp
+	rm -fr *.o
+
+clean: mostlyclean
+
+distclean: clean
+	rm -f Makefile Makefile.in POTFILES *.mo *.msg *.cat *.cat.m
+
+maintainer-clean: distclean
+	@echo "This command is intended for maintainers to use;"
+	@echo "it deletes files that may require special tools to rebuild."
+	rm -f $(GMOFILES)
+
+distdir = ../$(PACKAGE)-$(VERSION)/$(subdir)
+dist distdir: update-po $(DISTFILES)
+	dists="$(DISTFILES)"; \
+	for file in $$dists; do \
+	  ln $(srcdir)/$$file $(distdir) 2> /dev/null \
+	    || cp -p $(srcdir)/$$file $(distdir); \
+	done
+
+update-po: Makefile
+	$(MAKE) $(PACKAGE).pot
+	PATH=`pwd`/../src:$$PATH; \
+	cd $(srcdir); \
+	catalogs='$(CATALOGS)'; \
+	for cat in $$catalogs; do \
+	  cat=`basename $$cat`; \
+	  lang=`echo $$cat | sed 's/\$(CATOBJEXT)$$//'`; \
+	  mv $$lang.po $$lang.old.po; \
+	  echo "$$lang:"; \
+	  if $(MSGMERGE) $$lang.old.po $(PACKAGE).pot -o $$lang.po; then \
+	    rm -f $$lang.old.po; \
+	  else \
+	    echo "msgmerge for $$cat failed!"; \
+	    rm -f $$lang.po; \
+	    mv $$lang.old.po $$lang.po; \
+	  fi; \
+	done
+
+POTFILES: POTFILES.in
+	( if test 'x$(srcdir)' != 'x.'; then \
+	    posrcprefix='$(top_srcdir)/'; \
+	  else \
+	    posrcprefix="../"; \
+	  fi; \
+	  rm -f $@-t $@ \
+	    && (sed -e '/^#/d' -e '/^[ 	]*$$/d' \
+		    -e "s@.*@	$$posrcprefix& \\\\@" < $(srcdir)/$@.in \
+		| sed -e '$$s/\\$$//') > $@-t \
+	    && chmod a-w $@-t \
+	    && mv $@-t $@ )
+
+Makefile: Makefile.in.in ../config.status POTFILES
+	cd .. \
+	  && CONFIG_FILES=$(subdir)/$@.in CONFIG_HEADERS= \
+	       $(SHELL) ./config.status
+
+# Tell versions [3.59,3.63) of GNU make not to export all variables.
+# Otherwise a system limit (for SysV at least) may be exceeded.
+.NOEXPORT:
