Index: .cvsignore
===================================================================
RCS file: /cvs/gnome/ORBit/.cvsignore,v
retrieving revision 1.6.4.1
diff -u -u -r1.6.4.1 .cvsignore
--- .cvsignore	2000/09/29 23:34:02	1.6.4.1
+++ .cvsignore	2000/09/30 15:44:19
@@ -20,4 +20,4 @@
 install-sh
 mkinstalldirs
 ORBit.spec
-ORBit.pc
+ORBit-2.0.pc
Index: Makefile.am
===================================================================
RCS file: /cvs/gnome/ORBit/Makefile.am,v
retrieving revision 1.13.4.3
diff -u -u -r1.13.4.3 Makefile.am
--- Makefile.am	2000/09/30 02:09:43	1.13.4.3
+++ Makefile.am	2000/09/30 15:44:19
@@ -4,7 +4,7 @@
 
 bin_SCRIPTS=orbit-config
 
-EXTRA_DIST=ORBit.spec.in ORBit.m4 ORBit.pc.in
+EXTRA_DIST=ORBit.spec.in ORBit.m4 ORBit-2.0.pc.in
 
 dist-hook: ORBit.spec
 	mkdir $(distdir)/docs
@@ -18,4 +18,5 @@
 m4data_DATA=ORBit.m4
 
 pkgconfigdir = $(libdir)/pkgconfig
-pkgconfig_DATA= ORBit.pc
+pkgconfig_DATA = ORBit-2.0.pc
+
Index: ORBit-2.0.pc.in
===================================================================
RCS file: ORBit-2.0.pc.in
diff -N ORBit-2.0.pc.in
--- /dev/null	Tue May  5 16:32:27 1998
+++ ORBit-2.0.pc.in	Sat Sep 30 11:44:19 2000
@@ -0,0 +1,12 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+
+Name: ORBit
+Description: High-performance CORBA Object Request Broker.
+Version: @VERSION@
+Requires: glib-2.0
+Libs: -L${libdir} -lORBit -lIIOP -lORBitutil @LIBS@ -lm
+Cflags: -I${includedir}
Index: configure.in
===================================================================
RCS file: /cvs/gnome/ORBit/configure.in,v
retrieving revision 1.98.4.6
diff -u -u -r1.98.4.6 configure.in
--- configure.in	2000/09/29 23:34:02	1.98.4.6
+++ configure.in	2000/09/30 15:44:19
@@ -99,7 +99,7 @@
 dnl Checks for libraries.
 
 # Make sure that we have glib>=1.1.16
-AM_PATH_GLIB(1.2.1,,
+AM_PATH_GLIB_2_0(1.2.1,,
 AC_MSG_ERROR([
 *** GLIB 1.2.1 or better is required. The latest version of GLIB
 *** is always available from ftp://ftp.gtk.org/.]))
@@ -347,5 +347,5 @@
 test/Makefile
 test/idl/Makefile
 test/test-poa/Makefile
-ORBit.pc
+ORBit-2.0.pc
 ORBit.spec)
Index: orbit-config.in
===================================================================
RCS file: /cvs/gnome/ORBit/orbit-config.in,v
retrieving revision 1.7
diff -u -u -r1.7 orbit-config.in
--- orbit-config.in	1999/10/08 04:49:07	1.7
+++ orbit-config.in	2000/09/30 15:44:19
@@ -87,8 +87,8 @@
        	;;
 
     client|server)
-	the_libs="$the_libs -L$libdir -lORBit -lIIOP -lORBitutil `glib-config --libs` @LIBS@ -lm"
-	the_flags="$the_flags `glib-config --cflags` -I$includedir "
+	the_libs="$the_libs -L$libdir -lORBit -lIIOP -lORBitutil `glib-config-2.0 --libs` @LIBS@ -lm"
+	the_flags="$the_flags `glib-config-2.0 --cflags` -I$includedir "
 	;;
 
    --use-service=*)
Index: libIDL/.cvsignore
===================================================================
RCS file: /cvs/gnome/libIDL/.cvsignore,v
retrieving revision 1.8.4.1
diff -u -u -r1.8.4.1 .cvsignore
--- libIDL/.cvsignore	2000/09/29 23:35:50	1.8.4.1
+++ libIDL/.cvsignore	2000/09/30 15:44:19
@@ -28,4 +28,4 @@
 libIDL.m4
 Makefile.msc
 libIDL.spec
-libIDL.pc
+libIDL-2.0.pc
Index: libIDL/Makefile.am
===================================================================
RCS file: /cvs/gnome/libIDL/Makefile.am,v
retrieving revision 1.65.4.3
diff -u -u -r1.65.4.3 Makefile.am
--- libIDL/Makefile.am	2000/09/30 02:09:43	1.65.4.3
+++ libIDL/Makefile.am	2000/09/30 15:44:19
@@ -33,7 +33,7 @@
 			  Makefile.msc README.win32 \
 			  libIDL.def libIDLConf.sh.in \
 			  stamp-parser lexer.c parser.c parser.h \
-			  libIDL.pc.in
+			  libIDL-2.0.pc.in
 
 info_TEXINFOS		= libIDL.texi
 pkginclude_HEADERS	= IDL.h
@@ -73,5 +73,5 @@
 		&& mv libIDLConf.tmp libIDLConf.sh
 
 pkgconfigdir = $(libdir)/pkgconfig
-pkgconfig_DATA= libIDL.pc
+pkgconfig_DATA = libIDL-2.0.pc
 
Index: libIDL/configure.in
===================================================================
RCS file: /cvs/gnome/libIDL/configure.in,v
retrieving revision 1.54.4.1
diff -u -u -r1.54.4.1 configure.in
--- libIDL/configure.in	2000/09/29 23:35:50	1.54.4.1
+++ libIDL/configure.in	2000/09/30 15:44:20
@@ -82,7 +82,7 @@
 AC_CHECK_PROG(HAVE_GLIB_CONFIG, glib-config, yes, no)
 
 AC_SUBST(GLIB_VERSION_REQUIRED)
-AM_PATH_GLIB($GLIB_VERSION_REQUIRED)
+AM_PATH_GLIB_2_0($GLIB_VERSION_REQUIRED)
 
 AC_STDC_HEADERS
 AC_CHECK_HEADERS(stddef.h)
@@ -105,6 +105,7 @@
 libIDL.spec
 libIDL.pc
 IDL.h.new
+libIDL-2.0.pc
 ],[
 AC_UPDATE_IF_CHANGED(IDL.h, IDL.h.new)
 ])
Index: libIDL/libIDL-2.0.pc.in
===================================================================
RCS file: libIDL-2.0.pc.in
diff -N libIDL-2.0.pc.in
--- /dev/null	Tue May  5 16:32:27 1998
+++ libIDL-2.0.pc.in	Sat Sep 30 11:44:20 2000
@@ -0,0 +1,12 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+
+Name: libIDL
+Description: IDL parsing library
+Version: @VERSION@
+Requires: glib-2.0
+Libs: -L${libdir} -lIDL
+Cflags: -I${includedir}
