diff -U2 -r oaf/oafd/ac-corba.c HEAD/oaf/oafd/ac-corba.c
--- oaf/oafd/ac-corba.c	Wed Aug  9 06:46:00 2000
+++ HEAD/oaf/oafd/ac-corba.c	Thu Aug 31 01:22:32 2000
@@ -122,5 +122,4 @@
 		child->active_servers =
 			g_hash_table_new (g_str_hash, g_str_equal);
-		g_hash_table_freeze (child->active_servers);
 		for (i = 0; i < cache->_u.active_servers._length; i++)
 			g_hash_table_insert (child->active_servers,
@@ -128,5 +127,4 @@
 					     active_servers._buffer[i],
 					     GINT_TO_POINTER (1));
-		g_hash_table_thaw (child->active_servers);
 	} else
 		CORBA_free (cache);
@@ -164,10 +162,8 @@
 		child->time_list_pulled = time (NULL);
 		child->by_iid = g_hash_table_new (g_str_hash, g_str_equal);
-		g_hash_table_freeze (child->by_iid);
 		for (i = 0; i < child->list->_length; i++)
 			g_hash_table_insert (child->by_iid,
 					     child->list->_buffer[i].iid,
 					     &(child->list->_buffer[i]));
-		g_hash_table_thaw (child->by_iid);
 	}
 
diff -U2 -r oaf/oafd/od-load.c HEAD/oaf/oafd/od-load.c
--- oaf/oafd/od-load.c	Wed Aug  9 06:46:00 2000
+++ HEAD/oaf/oafd/od-load.c	Thu Aug 31 01:22:49 2000
@@ -321,10 +321,8 @@
 	retval = CORBA_sequence_OAF_ServerInfo_allocbuf (n);
 
-	g_hash_table_freeze (*by_iid);
 	for (i = 0, cur = entries; i < n; i++, cur = cur->next) {
 		memcpy (&retval[i], cur->data, sizeof (OAF_ServerInfo));
 		g_hash_table_insert (*by_iid, retval[i].iid, &retval[i]);
 	}
-	g_hash_table_thaw (*by_iid);
 
 	return retval;
