diff -U2 -r ORBit/configure.in ../HEAD/ORBit/configure.in
--- ORBit/configure.in	Mon Aug 28 12:53:19 2000
+++ ../HEAD/ORBit/configure.in	Thu Aug 31 00:46:01 2000
@@ -100,5 +100,5 @@
 
 # Make sure that we have glib>=1.1.16
-AM_PATH_GLIB(1.2.1,,
+AM_PATH_GLIB_2_0(1.3.1,,
 AC_MSG_ERROR([
 *** GLIB 1.2.1 or better is required. The latest version of GLIB
diff -U2 -r ORBit/libIDL/configure.in ../HEAD/ORBit/libIDL/configure.in
--- ORBit/libIDL/configure.in	Mon Aug 28 12:53:23 2000
+++ ../HEAD/ORBit/libIDL/configure.in	Mon Aug 28 13:22:30 2000
@@ -20,5 +20,5 @@
 LIBIDL_BINARY_AGE=8
 LIBIDL_LIBRARY_VERSION=$LIBIDL_MAJOR_VERSION.$LIBIDL_MINOR_VERSION.$LIBIDL_MICRO_VERSION
-GLIB_VERSION_REQUIRED=1.2.0
+GLIB_VERSION_REQUIRED=1.3.1
 AC_DIVERT_POP()dnl
 
@@ -80,8 +80,8 @@
 
 dnl Checks for libraries.
-AC_CHECK_PROG(HAVE_GLIB_CONFIG, glib-config, yes, no)
+AC_CHECK_PROG(HAVE_GLIB_CONFIG, glib-config-2.0, yes, no)
 
 AC_SUBST(GLIB_VERSION_REQUIRED)
-AM_PATH_GLIB($GLIB_VERSION_REQUIRED)
+AM_PATH_GLIB_2_0($GLIB_VERSION_REQUIRED)
 
 AC_STDC_HEADERS
diff -U2 -r ORBit/libIDL/util.c ../HEAD/ORBit/libIDL/util.c
--- ORBit/libIDL/util.c	Mon Aug 28 12:53:23 2000
+++ ../HEAD/ORBit/libIDL/util.c	Tue Aug 29 14:31:03 2000
@@ -504,7 +504,7 @@
 	if (__IDL_cur_filename) {
 #ifdef HAVE_CPP_PIPE_STDIN
-		filename = __IDL_cur_filename;
+		filename = g_strdup (__IDL_cur_filename);
 #else
-		filename = g_basename (__IDL_cur_filename);
+		filename = g_path_get_basename (__IDL_cur_filename);
 #endif
 	} else
@@ -526,4 +526,6 @@
 			fprintf (stderr, "Error: %s\n", s);
 	}
+
+	g_free (filename);
 }
 
@@ -539,7 +541,7 @@
 	if (__IDL_cur_filename) {
 #ifdef HAVE_CPP_PIPE_STDIN
-		filename = __IDL_cur_filename;
+		filename = g_strdup (__IDL_cur_filename);
 #else
-		filename = g_basename (__IDL_cur_filename);
+		filename = g_path_get_basename (__IDL_cur_filename);
 #endif
 	} else
@@ -556,4 +558,6 @@
 			fprintf (stderr, "Warning: %s\n", s);
 	}
+
+	g_free (filename);
 }
 
diff -U2 -r ORBit/orbit-config.in ../HEAD/ORBit/orbit-config.in
--- ORBit/orbit-config.in	Mon Aug 28 12:53:19 2000
+++ ../HEAD/ORBit/orbit-config.in	Mon Aug 28 13:23:07 2000
@@ -88,6 +88,6 @@
 
     client|server)
-	the_libs="$the_libs -L$libdir -lORBit -lIIOP -lORBitutil `glib-config --libs` @LIBS@ -lm"
-	the_flags="$the_flags `glib-config --cflags` -I$includedir "
+	the_libs="$the_libs -L$libdir -lORBit -lIIOP -lORBitutil `glib-config-2.0 --libs` @LIBS@ -lm"
+	the_flags="$the_flags `glib-config-2.0 --cflags` -I$includedir "
 	;;
 
diff -U2 -r ORBit/src/IIOP/connection.c ../HEAD/ORBit/src/IIOP/connection.c
--- ORBit/src/IIOP/connection.c	Mon Aug 28 12:53:20 2000
+++ ../HEAD/ORBit/src/IIOP/connection.c	Thu Aug 31 01:25:53 2000
@@ -106,5 +106,5 @@
 
 #if defined(HAVE_HOSTS_ACCESS) && defined (HAVE_TCPD_H)
-  argv0_val = g_strdup(g_basename(argv0)); /* For TCP wrappers */
+  argv0_val = g_path_get_basename(argv0); /* For TCP wrappers */
 #endif
 
diff -U2 -r ORBit/src/idl-compiler/orbit-c-backend.c ../HEAD/ORBit/src/idl-compiler/orbit-c-backend.c
--- ORBit/src/idl-compiler/orbit-c-backend.c	Mon Aug 28 12:53:20 2000
+++ ../HEAD/ORBit/src/idl-compiler/orbit-c-backend.c	Thu Aug 31 01:26:34 2000
@@ -31,5 +31,5 @@
   }
   
-  base_filename = g_basename(base_filename);
+  base_filename = g_path_get_basename(base_filename);
 
   if(!disable_headers) {
@@ -48,5 +48,5 @@
   }
   g_string_sprintf(curfile_name, "%s.h", base_filename);
-  hdrname = g_strdup(g_basename(curfile_name->str));
+  hdrname = g_path_get_basename(curfile_name->str);
 
   if(!disable_common) {
@@ -112,4 +112,5 @@
   g_string_free(curfile_name, TRUE);
   g_free(hdrname);
+  g_free(base_filename);
 }
 
diff -U2 -r ORBit/src/orbit-idl-compiler/backends/c/orbit-idl-c-backend.c ../HEAD/ORBit/src/orbit-idl-compiler/backends/c/orbit-idl-c-backend.c
--- ORBit/src/orbit-idl-compiler/backends/c/orbit-idl-c-backend.c	Mon Aug 28 12:53:20 2000
+++ ../HEAD/ORBit/src/orbit-idl-compiler/backends/c/orbit-idl-c-backend.c	Thu Aug 31 01:27:07 2000
@@ -16,5 +16,5 @@
   OIDL_C_Info ci;
 
-  ci.base_name = g_strdup(g_basename(rinfo->input_filename));
+  ci.base_name = g_path_get_basename(rinfo->input_filename);
   ctmp = strrchr(ci.base_name, '.');
   g_assert(ctmp);
@@ -61,7 +61,11 @@
   char *ctmp;
   char *cmdline;
+  char *tmpbasename;
 
   basein = alloca(strlen(input_filename) + sizeof("-skelimpl.c"));
-  strcpy(basein, g_basename(input_filename));
+
+  tmpbasename = g_path_get_basename (input_filename);
+  strcpy(basein, tmpbasename);
+  g_free (tmpbasename);
 
   ctmp = strrchr(basein, '.');
diff -U2 -r ORBit/src/orbit-idl-compiler/orbit-idl-main.c ../HEAD/ORBit/src/orbit-idl-compiler/orbit-idl-main.c
--- ORBit/src/orbit-idl-compiler/orbit-idl-main.c	Mon Aug 28 12:53:21 2000
+++ ../HEAD/ORBit/src/orbit-idl-compiler/orbit-idl-main.c	Thu Aug 31 00:46:01 2000
@@ -156,5 +156,7 @@
   /* Do it */
   while((arg=poptGetArg(pcon))!=NULL) {
-    rinfo.input_filename = g_basename(arg);
+
+    /* Possible memory leak */
+    rinfo.input_filename = g_path_get_basename(arg);
     if (!orbit_idl_to_backend(arg, &rinfo)) {
       g_warning("%s compilation failed", arg);
